flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -237,13 +237,14 @@ public TaskExecutor(
 			TaskManagerServices taskExecutorServices,
 			HeartbeatServices heartbeatServices,
 			TaskManagerMetricGroup taskManagerMetricGroup,
-			@Nullable String metricQueryServiceAddress,
+			String metricQueryServiceAddress,
 			BlobCacheService blobCacheService,
 			FatalErrorHandler fatalErrorHandler,
 			PartitionTable<JobID> partitionTable) {
 
 		super(rpcService, AkkaRpcServiceUtils.createRandomName(TASK_MANAGER_NAME));
 
+		Preconditions.checkNotNull(metricQueryServiceAddress);
```

Comment:
IIRC you can change 

```java
this.metricQueryServiceAddress = metricQueryServiceAddress;
```

to

```java
this.metricQueryServiceAddress = checkNotNull(metricQueryServiceAddress);
```

instead of a single line.